src/main.rs

```diff
@@ -116,14 +126,21 @@ fn main() {
         });
     }
 
-    let sleep_duration = std::time::Duration::new(config.run_interval, 0);
     let bitbucket = bitbucket::Bitbucket::new(&config.bitbucket, &fanout);
     if let Some(t) = config.clone().telegram {
         if t.enabled {
             t.announce_from(fanout.subscribe()).expect("Failed to authenticate with Telegram");
         }
     }
 
+    let mut sleep_duration: std::time::Duration;
+    let mut schedule: Option<CronSchedule> = None;
+
+    match &config.run_interval {
```

Comment:
You can remove some of the references to make it look nicer. I've also changed some of the `let` bindings that will make more sense for the code below.

```rust
let mut fixed_interval: Option<std::time::Duration> = None;
let mut schedule: Option<CronSchedule> = None;

match config.run_interval {
    Interval::Cron { expression } => {
        schedule = Some(CronSchedule::parse(expression).unwrap())
    },
    Interval::Fixed { interval } => {
        fixed_interval = Some(std::time::Duration::new(interval, 0))
    }
};

```